products/revenue_analytics/backend/models.py

```diff
@@ -88,27 +91,112 @@ def for_schema_source(source: ExternalDataSource) -> Optional["RevenueAnalyticsR
         team = table.team
         revenue_config = team.revenue_config
 
-        zero_decimal_currencies = ", ".join([f"'{currency.value}'" for currency in ZERO_DECIMAL_CURRENCIES_IN_STRIPE])
         base_currency = (revenue_config.baseCurrency or DEFAULT_CURRENCY).value
 
-        query = f"""
-    SELECT
-        id AS id,
-        created_at AS timestamp,
-        toDecimal(amount, {EXCHANGE_RATE_DECIMAL_PRECISION}) AS original_amount,
-        upper(currency) AS original_currency,
-        original_currency IN ({zero_decimal_currencies}) AS currency_is_zero_decimal,
-        if(currency_is_zero_decimal, toDecimal(1, {EXCHANGE_RATE_DECIMAL_PRECISION}), toDecimal(100, {EXCHANGE_RATE_DECIMAL_PRECISION})) AS amount_decimal_divider,
-        divideDecimal(original_amount, amount_decimal_divider) AS adjusted_original_amount,
-        '{base_currency}' AS currency,
-        convertCurrency(original_currency, currency, adjusted_original_amount, _toDate(ifNull(timestamp, toDateTime(0)))) AS amount
-    FROM
-        {table.name}
-"""
+        # Even though we need a string query for the view,
+        # using an ast allows us to comment what each field means, and
+        # avoid manual interpolation of constants, leaving that to the HogQL printer
+        query = ast.SelectQuery(
+            select=[
+                # Base fields to allow insights to work (need `distinct_id` AND `timestamp` fields)
+                ast.Alias(alias="id", expr=ast.Field(chain=["id"])),
+                ast.Alias(alias="timestamp", expr=ast.Field(chain=["created_at"])),
+                # Compute the original amount in the original currency
```

Comment:
I'd love to add some empty lines in here, but ruff keeps formatting them back, I'll see if I can tweak the config